packages/grid/x-data-grid/src/components/GridRowCount.tsx

```diff
@@ -2,7 +2,7 @@ import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
 import { unstable_composeClasses as composeClasses } from '@mui/utils';
-import { styled, SxProps, Theme } from '@mui/material/styles';
+import { styled, SxProps, Theme } from '@mui/system';
```

Comment:
Based on https://github.com/mui/material-ui/issues/35840, could we use nested imports here ?

Reply:
`styled` is exported directly from the index. It does not have a nested folder after the build. We could improve it on the core but it should not block this PR.

Both of these are the same (in terms of efficiency).
```js
import { styled } from '@mui/material/styles';
import { styled } from '@mui/system';
```